core/LSU.cpp

```diff
@@ -364,7 +382,15 @@ namespace olympia
             }
         }
 
-        out_mmu_lookup_req_.send(mem_access_info_ptr);
+        // TODO: set a max upper bound
+        uint32_t mmu_sync_delay = 0;
+        while (out_mmu_lookup_req_.isDriven(mmu_sync_delay))
+        {
+            mmu_sync_delay++;
+        }
```

Comment:
I kinda know what you're doing here -- and I kinda don't.  :laughing:   Are you looking to see how many outstanding requests there are?

Reply:
My thoughts on the synchronization problem was that multiple ldst pipelines could send requests at the same time

Reply:
Ok, so you're saying they can _make requests_ at the same time, but the requests are staggered coming into the MMU.  

I think this will break the LSU pipeline concept.  Specifically, if a load is traversing the LS pipeline and the pipeline is "programmed" to stage it in 4 cycles: 

1. agen
2. xlate
3. cache
4. drive

You're artificially delaying stage 2 with a late request to the MMU, so when the load hits the cache stage of the pipeline, it will see the PA is not ready and drop the load: https://github.com/riscv-software-src/riscv-perf-model/blob/33d2e2fbc99a3439b5d15ee9220558ca1467d909/core/LSU.cpp#L437